src/index-file/meta.ts

```diff
@@ -23,54 +23,69 @@ export async function readFileMeta(buffer: ArrayBuffer): Promise<FileMeta> {
 	const format = dataView.getUint8(1);
 
 	if (format !== FileFormat.CSV && format !== FileFormat.JSONL) {
-		throw new Error(`unexpected file format. Got: ${format}`)
+		throw new Error(`unexpected file format. Got: ${format}`);
 	}
 
 	const readOffset = dataView.getBigUint64(2);
 
 	return {
 		version,
 		format,
-		readOffset
-	}
+		readOffset,
+	};
 }
 
-
-
 export type IndexMeta = {
 	fieldName: string;
-	fieldType: bigint;
+	fieldType: number;
 };
 
-export async function unmarshalBinaryForIndexMeta(
-	resolver: RangeResolver,
-	buffer: ArrayBuffer
+export type IndexHeader = {
```

Comment:
Is this all duplicating the code in the other file?

Reply:
Yes but also `IndexHeader` is suppose to condense `IndexMeta[]` since for a given `IndexMeta`, there exists more than one element. `IndexHeader[]` ensures every `IndexHeader` has a unique `fieldName` and a list of it's `fieldType`.